tasks/notify.py

```diff
@@ -511,6 +514,33 @@ def has_upcoming_notifies_according_to_redis(
             return True
         return False
 
+    @sentry_sdk.trace
+    def save_patch_totals(self, comparison: ComparisonProxy) -> None:
+        """Saves patch coverage to the CompareCommit, if it exists.
+        This is done to make sure the patch coverage reported by notifications and UI is the same
+        (because they come from the same source)
+        """
+        if comparison.project_coverage_base.commit is None:
+            # This is the base that will be saved in the CommitComparison
+            # Even if the patch coverage could come from a different commit
+            return
```

Comment:
in other words, we identify a `CommitComparison` by its project coverage base irrespective of whether its patch coverage base is different?

Reply:
Yes, that's right... but this was a good callout.
The Comparison is created by the `UploadFinisher` task [here](https://github.com/codecov/worker/blob/main/tasks/upload_finisher.py#L322). I thought that `Pull.compared_to` was the project base. But maybe that's not the case... I'll review before merging

Reply:
Tracking the code to [this](https://github.com/codecov/worker/blob/68d5a8930db22d978081aa6e388351083fe1ccd1/tasks/notify.py#L334) (where we decide what is the base_commit for the comparison) it seems that it's always the project base, unless there is no pull, then we use the commit's parent

So I guess the logic is fine. If we have a project base we would have created a comparison. Otherwise there is no Pull object and so there's no Comparison.

But this also means we might emit patch coverage in the UI 🤔
Should we then `get_or_create` a comparison instead of only updating if it exists... it might be a better idea